src/App.js

```diff
@@ -80,32 +81,39 @@ const App = (props) => {
         } catch (e) {
             chrome.hideGlobalFilter();
         }
+
+        setHasSystemsReady(true);
     }
 
     useEffect(() => {
         initChrome();
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
+    if (!arePermissionsReady || !hasSystemsReady) {
+        return <PageLoading />;
+    }
+
+    if (hasSystemsReady && !hasSystems) {
+        return <ZeroState />;
```

Comment:
The `AppZeroState` component has a `hasSystems` prop and a function and allows to wrap children and depending on that it shows a loading state or the children. It would also separate the concerns of the `App` and `ZeroState` component better.

Reply:
Nice idea, we need to migrate to the new implementation indeed. Will make the changes and push soon

Reply:
I have found out that the new implementation does not include the changes necessary for the dashboard app. This maybe because the dashboard zero-state is very different from other standard zero-states.

Reply:
@mkholjuraev We can still just build a similar wrapper component, or?

Reply:
I was able to workaround this ^